projects/storefrontlib/src/cms-components/checkout/services/express-checkout.service.ts

```diff
@@ -0,0 +1,118 @@
+import { Injectable } from '@angular/core';
+import { combineLatest, of } from 'rxjs';
+import { filter, map, switchMap, withLatestFrom } from 'rxjs/operators';
+
+import {
+  Address,
+  CheckoutDeliveryService,
+  UserAddressService,
+  UserPaymentService,
+  PaymentDetails,
+  DeliveryMode,
+  CheckoutPaymentService,
+} from '@spartacus/core';
+import { CheckoutConfigService } from './checkout-config.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ExpressCheckoutService {
+  shippingAddress$;
+  deliveryMode$;
+  paymentMethod$;
+
+  constructor(
+    protected userAddressService: UserAddressService,
+    protected checkoutDeliveryService: CheckoutDeliveryService,
+    protected checkoutPaymentService: CheckoutPaymentService,
+    protected userPaymentService: UserPaymentService,
+    protected checkoutConfigService: CheckoutConfigService
+  ) {
+    this.setShippingAddress();
+    this.setDeliveryMode();
+    this.setPaymentMethod();
+  }
+
+  protected setShippingAddress() {
+    this.userAddressService.loadAddresses();
+    this.shippingAddress$ = this.userAddressService
+      .getAddressesLoadedSuccess()
+      .pipe(
+        filter(Boolean),
+        switchMap(() => this.userAddressService.getAddresses()),
+        map((addresses: Address[]) => {
+          const defaultAddress =
+            addresses.find(address => address.defaultAddress) || addresses[0];
+          if (defaultAddress) {
+            this.checkoutDeliveryService.setDeliveryAddress(defaultAddress);
+          }
+          return Boolean(defaultAddress);
+        })
+      );
+  }
+
+  protected setPaymentMethod() {
+    this.userPaymentService.loadPaymentMethods();
+    this.paymentMethod$ = this.userPaymentService
+      .getPaymentMethodsLoadedSuccess()
+      .pipe(
+        filter(Boolean),
+        switchMap(() => this.userPaymentService.getPaymentMethods()),
+        map((paymentMethods: PaymentDetails[]) => {
+          const defaultPayment =
+            paymentMethods.find(
+              paymentMethod => paymentMethod.defaultPayment
+            ) || paymentMethods[0];
+          if (defaultPayment) {
+            this.checkoutPaymentService.setPaymentDetails(defaultPayment);
```

Comment:
Same here. You could get redirected to review step to quickly and guards in review page would redirect you back to the payment step.